src/logic/cookies/utils/index.ts

```diff
@@ -35,16 +30,19 @@ export const saveCookie = async <T extends Record<string, any>>(
   const prefix = withoutPrefix ? '' : PREFIX
   try {
     const stringifiedValue = JSON.stringify(value)
-    await Cookies.set(`${prefix}${key}`, stringifiedValue, options)
+    Cookies.set(`${prefix}${key}`, stringifiedValue, options)
   } catch (err) {
     logError(Errors._701, `cookie ${key} – ${err.message}`)
   }
 }
 
-export const removeCookie = (key: string, path: string, domain: string): void => Cookies.remove(key, { path, domain })
+export const removeCookies = (nameRegex: RegExp): void => {
+  const allCookies = Cookies.get()
+  const matching = Object.keys(allCookies).filter((name) => nameRegex.test(name))
 
-export const removeCookies = (cookieList: Cookie[]): void => {
-  // Extracts domain, e.g. gnosis-safe.io
-  const domain = location.host.split('.').slice(-2).join('.')
-  cookieList.forEach((cookie) => removeCookie(cookie.name, cookie.path, `.${domain}`))
+  try {
+    matching.forEach((name) => Cookies.remove(name, JSON.parse(allCookies[name])))
```

Comment:
This would fail if `allCookies[name]` is not a stringified json which is the case for some stored cookies

Reply:
We could remove the cookies just by their name without passing in a second param.

Reply:
For reference:
![image](https://user-images.githubusercontent.com/20442784/156559366-769992ea-be03-4906-8c7e-e810915b8c45.png)

Reply:
I've reverted this.